libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -108,6 +128,30 @@ private static PolicyManager createPolicyManager() {
         return new PolicyManager(serverPolicy, agentEntitlements, pluginPolicies, resolver, ENTITLEMENTS_MODULE);
     }
 
+    static void addInstrumentationMethod(
+        Map<MethodKey, CheckMethod> checkMethods,
+        Class<?> clazz,
```

Comment:
Since there are two other `Class` parameters, could we give this one a more meaningful name? I think it might be something like `targetSuperclass` perhaps?